code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -41,282 +40,190 @@ internal sealed class AEPPushTemplate {
     internal val body: String
 
     // Required, Version of the payload assigned by the authoring UI.
-    internal var payloadVersion: Int?
-        private set
+    internal val payloadVersion: String
```

Comment:
This'll still be backward compatible right (Changing type from Int to String) the configuration currently done is in String or Int ?

Reply:
Reading the specs in wiki.. Looks like the payloadVesion is a String.

We can check with the Campaign team and may be have them follow the [semantic versioning ](https://semver.org/#semantic-versioning-200) to accurately represent version and their changes.


As far as backward compatibility goes, I am not worried about that yet. Since there is no versioning enforced yet and if so enforced it will be more a client side logic to handle differences in the extraction payload for different versions.